placenames.mss

```diff
@@ -307,6 +319,10 @@
     text-wrap-width: 70; // 5.0 em
     text-line-spacing: -0.70; // -0.05 em
     text-margin: 9.8; // 0.7 em
+    text-placement-type: simple;
+    text-placements: "N,S,E,W,NW,NE,SE";
```

Comment:
Did you try adding the "X" position as well?

Reply:
Yes, I did, but this screwed up the placement. As far as I was able to determine, if there is an X in the placement list it will try placement on its center and then abandon placement if there is overlap. It will not try N, S, E, W, etc...

But this was hard to test well and there is little to no documentation from Mapnik.

Reply:
Indeed. As far as I understand, the issue is actually with `text-dx`/`text-dy` which apply including in the X placement case... and Mapnik doc is really scarce on this.

Digging into Mapnik code to try to have a better understanding, I discovered the "C" placement which is totally undocumented and seems to do magic.

Here is the current view in CyclOSM:
![old](https://user-images.githubusercontent.com/3856586/82209848-4423a500-990e-11ea-956e-7191386bf00b.png)

And the one with the leading "C" placement:
![new](https://user-images.githubusercontent.com/3856586/82209861-4ab21c80-990e-11ea-8f5e-6a2edbbac7d2.jpg)

It seems to have both benefits from your changes and stricter positionning (look at Nijverdal position, West, which is almost unaffected by these changes).

So, I merged your PR adding this extra commit https://github.com/cyclosm/cyclosm-cartocss-style/commit/a50958ac5ffab263af67d25944e3ee6019195209. I'm welcoming feedbacks if you think I missed something with the C placement.

Reply:
Very well, thanks for the addition. Indeed I did not find `C` placement.

(Somewhere deep on my list is to test and improve the Mapnik 3 documentation. That would benefit map creators, including me, a lot)